File: src/main/java/architecture/goldenboughs_lib/mixed/geckolib/IAnimationController.java
package architecture.goldenboughs_lib.mixed.geckolib;

import software.bernie.geckolib.animatable.GeoAnimatable;
import software.bernie.geckolib.animation.AnimationController;

public interface IAnimationController<T extends GeoAnimatable> {
	static <T extends GeoAnimatable> IAnimationController<T> of(AnimationController<T> animatable) {
		//noinspection unchecked
		return (IAnimationController<T>) animatable;
	}

	/**
	 * 如果 lockBones 和 enabledBones 都不为空则先判断 enabledBones 在判断 lockBones
	 *
	 * @return 该骨骼是否可以影响影响
	 */
	boolean goldenboughs_lib$isInfluence(String boneName);

	AnimationController<T> goldenboughs_lib$lockBones(String... boneNames);

	AnimationController<T> goldenboughs_lib$removeLockBones(String... boneNames);

	AnimationController<T> goldenboughs_lib$enabledBones(String... boneNames);

	AnimationController<T> goldenboughs_lib$removeEnabledBones(String... boneNames);

	boolean goldenboughs_lib$isBoneExist(String boneName);
}

File: src/main/kotlin/architecture/goldenboughs_lib/mixed/geckolib/IAnimationController.kt
package architecture.goldenboughs_lib.mixed.geckolib

import software.bernie.geckolib.animatable.GeoAnimatable
import software.bernie.geckolib.animation.AnimationController

interface IAnimationController<T : GeoAnimatable> {
	/**
	 * 如果 lockBones 和 enabledBones 都不为空则先判断 enabledBones 在判断 lockBones
	 *
	 * @return 该骨骼是否可以影响影响
	 */
	fun `goldenboughs_lib$isInfluence`(boneName: String): Boolean

	fun `goldenboughs_lib$lockBones`(vararg boneNames: String): AnimationController<T>

	fun `goldenboughs_lib$removeLockBones`(vararg boneNames: String): AnimationController<T>

	fun `goldenboughs_lib$enabledBones`(vararg boneNames: String): AnimationController<T>

	fun `goldenboughs_lib$removeEnabledBones`(vararg boneNames: String): AnimationController<T>

	fun `goldenboughs_lib$isBoneExist`(boneName: String): Boolean

	companion object {
		@JvmStatic
		fun <T : GeoAnimatable> of(animatable: AnimationController<T>): IAnimationController<T> {
			return animatable as IAnimationController<T>
		}
	}
}
